optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -198,474 +201,160 @@
-                `True` if the device is CUDA, otherwise defaults to `False`.
-            model_save_dir (`str`, *optional*, defaults to `""`):
-                The directory under which the model exported to ONNX was saved.
-            last_encoder_model_name (`str`, *optional*, defaults to `optimum.onnxruntime.utils.ONNX_ENCODER_NAME`):
-                The name of the ONNX file containing the encoder part of the model.
-            last_decoder_model_name (`str`, *optional*, defaults to `optimum.onnxruntime.utils.ONNX_DECODER_NAME`):
-                The name of the ONNX file containing the decoder part of the model.
-            last_decoder_with_past_model_name (`str`, *optional*, defaults to `optimum.onnxruntime.utils.ONNX_DECODER_WITH_PAST_NAME`):
-                The name of the ONNX file containing the decoder with past key/values part of the model.
-        """
-        ABC.__init__(self)
+        self.session = session
+        self.config = config
+        self._device = device
+        self.use_io_binding = use_io_binding
+        self.main_input_name = main_input_name
+        self.normalized_config = NormalizedConfigManager.get_normalized_config_class(self.config.model_type)(
+            self.config
+        )
+        self.input_names = {input_key.name: idx for idx, input_key in enumerate(self.session.get_inputs())}
+        self.output_names = {output_key.name: idx for idx, output_key in enumerate(self.session.get_outputs())}
+        self.name_to_np_type = TypeHelper.get_io_numpy_type_map(self.session) if self.use_io_binding else None
 
-        self.encoder_file_name = last_encoder_model_name
-        self.decoder_file_name = last_decoder_model_name
-        self.decoder_file_with_past_name = last_decoder_with_past_model_name
+    def prepare_output_buffer(self, batch_size, sequence_length):
+        """Prepare the buffer of output(`last_hidden_state`) with a 1D tensor on shape: (batch_size, sequence_length, hidden_size)."""
+        ort_type = TypeHelper.get_output_type(self.session, "last_hidden_state")
+        torch_type = TypeHelper.ort_type_to_torch_type(ort_type)
 
-        self.config = config
+        hidden_size = self.normalized_config.hidden_size
+        output_shape = (batch_size, sequence_length, hidden_size)
+        output_buffer = torch.empty(np.prod(output_shape), dtype=torch_type, device=self._device).contiguous()
 
-        self.use_io_binding = use_io_binding
-        self.model_save_dir = model_save_dir
+        return output_shape, output_buffer
 
-        self.providers = encoder_session.get_providers()
-        self._device = get_device_for_provider(encoder_session.get_providers()[0])
+    def prepare_io_binding(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+    ):
+        io_binding = self.session.io_binding()
 
-        if "TensorrtExecutionProvider" in self.providers and self.use_io_binding:
```

Comment:
Is this removed? I could not find it elsewhere. If removed, why?

Reply:
It is in `ORTModel`, now `ORTModelForConditionalGeneration` uses the base class initialiazer, so we do not need that anymore here.